src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -181,15 +193,20 @@ public void rollback() throws IOException {
     }
 
     public void close() throws IOException, DataException {
-        if (timer != null) {
-            timer.cancel();
-        }
+        stop();
 
         // Flush last file, updating index
         finishFile(true);
+    }
+
+    public void stop() {
+        stopped = true;
 
-        // Setting to null so subsequent calls to close won't write it again
-        currentFile = null;
+        if (timer != null) {
```

Comment:
If this is called in parallel - timer could turn into null between the check and use, and you'd have an NPE

Reply:
How can you be sure?  
Remember that you also call it from TopicPartitionWriter.stop and TopicPartitionWriter.close. Can these also not be called in parallel? and what about what calls them?

Reply:
i guess i can never be sure but the logic is - some kafka thread accepts a command to lose partition - then it calls stop here . So unless two such commands are sent somehow it shouldnt happen